src/widgets/allergies/allergies-overview.component.tsx

```diff
@@ -1,84 +1,115 @@
-import HorizontalLabelValue from "../../ui-components/cards/horizontal-label-value.component";
+import React, { useState, useEffect } from "react";
+import { capitalize } from "lodash-es";
+import { useTranslation } from "react-i18next";
 import { createErrorHandler } from "@openmrs/esm-error-handling";
 import { useCurrentPatient } from "@openmrs/esm-api";
-import AllergyForm from "./allergy-form.component";
 import { openWorkspaceTab } from "../shared-utils";
 import useChartBasePath from "../../utils/use-chart-base";
-import { useTranslation } from "react-i18next";
+import EmptyState from "../../ui-components/empty-state/empty-state.component";
+import HorizontalLabelValue from "../../ui-components/cards/horizontal-label-value.component";
+import SummaryCard from "../../ui-components/cards/summary-card.component";
+import SummaryCardRow from "../../ui-components/cards/summary-card-row.component";
+import SummaryCardFooter from "../../ui-components/cards/summary-card-footer.component";
+import {
+  performPatientAllergySearch,
+  Allergy
+} from "./allergy-intolerance.resource";
+import AllergyForm from "./allergy-form.component";
+import styles from "./allergies-overview.css";
 
 export default function AllergiesOverview(props: AllergiesOverviewProps) {
-  const [patientAllergies, setPatientAllergies] = React.useState(null);
-  const [
-    isLoadingPatient,
-    patient,
-    patientUuid,
-    patientErr
-  ] = useCurrentPatient();
+  const initialAllergiesBatchCount = 3;
+  const [allPatientAllergies, setAllPatientAllergies] = useState<Allergy[]>(
+    null
+  );
+  const [initialAllergiesBatch, setInitialAllergiesBatch] = useState<Allergy[]>(
+    []
+  );
+  const [allergiesExpanded, setAllergiesExpanded] = useState<boolean>(false);
+  const [isLoadingPatient, patient, patientUuid] = useCurrentPatient();
   const chartBasePath = useChartBasePath();
   const allergiesPath = chartBasePath + "/" + props.basePath;
   const { t } = useTranslation();
 
-  React.useEffect(() => {
-    if (patient) {
-      const abortController = new AbortController();
+  useEffect(() => {
+    if (!isLoadingPatient && patient) {
+      const sub = performPatientAllergySearch(
+        patient.identifier[0].value
+      ).subscribe(allergies => {
+        setAllPatientAllergies(allergies);
+        setInitialAllergiesBatch(
+          allergies.slice(0, initialAllergiesBatchCount)
+        );
+      }, createErrorHandler());
 
-      performPatientAllergySearch(patient.identifier[0].value, abortController)
-        .then(allergies => setPatientAllergies(allergies.data))
-        .catch(createErrorHandler());
+      return () => sub.unsubscribe();
+    }
+  }, [isLoadingPatient, patient]);
 
-      return () => abortController.abort();
+  useEffect(() => {
+    if (allPatientAllergies?.length <= initialAllergiesBatchCount) {
+      setAllergiesExpanded(true);
     }
-  }, [patient]);
+  }, [allPatientAllergies]);
+
+  const loadMoreAllergies = () => {
```

Comment:
I would call this `showMoreAllergies`

```suggestion
  const showMoreAllergies = () => {
```